src/components/tl/sharp/tl_sharp_team.c

```diff
@@ -42,21 +41,24 @@ UCC_CLASS_INIT_FUNC(ucc_tl_sharp_team_t, ucc_base_context_t *tl_context,
         self->oob_ctx.oob = &UCC_TL_TEAM_OOB(self);
     }
 
-    if (sharp_ctx == NULL) {
-        status = ucc_topo_init(set, ctx->super.super.ucc_context->topo, &topo);
-        if (UCC_OK != status) {
-            tl_error(ctx->super.super.lib, "failed to init team topo");
-            return status;
-        }
+    status = ucc_topo_init(set, ctx->super.super.ucc_context->topo, &self->topo);
+    if (UCC_OK != status) {
+        tl_error(ctx->super.super.lib, "failed to init team topo");
+        return status;
+    }
+
+    if (ucc_topo_max_ppn(self->topo) != 1) {
+        tl_info(ctx->super.super.lib, "sharp team not supported with ppn > 1");
+        return UCC_ERR_NOT_SUPPORTED;
```

Comment:
when return from team create function with error status you need to cleanup all resources allocated since destructor will not be called. Here need to free ucc_topo